client/src/components/v3/dashboard/V3SpeedDistanceChart.tsx

```diff
@@ -0,0 +1,62 @@
+import React, { useEffect, useRef, useState } from 'react';
+
+import { Sensor, useSensorData } from 'api/common/data';
+import { useChannel } from 'api/common/socket';
+import SpeedDistanceChart from 'components/common/charts/SpeedDistanceChart';
+import { ChartPoint } from 'types/chart';
+import { ReedDistanceRT, ReedVelocityRT } from 'types/data';
+
+export const V3SDChartKey = 'v3-dashboard-speed-distance-chart-data';
+
+/**
+ * Passes V3 data to the speed distance chart component
+ *
+ * @returns Component
+ */
+export function V3SpeedDistanceChart() {
+  const storedData = sessionStorage.getItem(V3SDChartKey);
+
+  // Used to store the data points
+  const [data, setStateData] = useState<ChartPoint[]>(
+    storedData ? JSON.parse(storedData) : [],
+  );
+  const maxPoint = useRef(-1);
+
+  // Reset when start message received
+  const reset = () => setStateData([]);
+  useChannel('module-3-start', reset);
+
+  // Store data for session
+  const setData = (newData: ChartPoint[]) => {
+    sessionStorage.setItem(V3SDChartKey, JSON.stringify(newData));
+    setStateData(newData);
+  };
+
+  // Speed
+  const point = useSensorData(3, Sensor.ReedVelocity, ReedVelocityRT);
+  const distance = useSensorData(3, Sensor.ReedDistance, ReedDistanceRT);
+
+  // Update data whenever the point is updated
+  useEffect(() => {
+    // Add new data point
+    if (
+      point &&
+      distance && // Non null
+      // New distance measurement
+      distance !== data[data.length - 1]?.y
+    ) {
+      maxPoint.current = Math.max(maxPoint.current, point);
+      setData([...data, { x: distance, y: point }]);
+    }
+    // Omit data in deps as otherwise there would be an infinite render loop
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [point, distance]);
```

Comment:
Just realised this now - we're actually putting two datapoints on the chart for every update, once when the speed changes, and again when the distance changes. If we remove distances from the dependency array (there might be a better way to do this though?) we get a much smoother looking graph:
![image](https://user-images.githubusercontent.com/17876556/111861378-e5054580-89a1-11eb-8aae-0b442284b72b.png)

Reply:
The only thing would be that if we had the same speed, the graph wouldn't update - though that's probably (?) not a very common occurrence.
Removed `distance` dependency.
299a0b5